src/sync/clients.ts

```diff
@@ -92,3 +95,55 @@ export async function setClient(
   clients.set(id, client);
   return setClients(clients, dagWrite);
 }
+
+export async function initClient(
+  dagStore: dag.Store,
+): Promise<[ClientID, Client]> {
+  return dagStore.withWrite(async dagWrite => {
+    const clients = await getClients(dagWrite);
+    const newClientID = makeUuid();
+    const bootstrapClient = [...clients.values()].reduce((prev, curr) =>
+      !prev || prev.heartbeatTimestampMs < curr.heartbeatTimestampMs
+        ? curr
+        : prev,
+    );
+
+    let newClientCommit;
+    if (bootstrapClient) {
+      const bootstrapCommit = await Commit.baseSnapshot(
+        bootstrapClient.headHash,
+        dagWrite,
+      );
+      // Copy the snapshot with one change: set last mutation id to 0.
+      newClientCommit = newSnapshot(
+        dagWrite.createChunk,
+        bootstrapCommit.meta.basisHash,
+        0 /* lastMutationID */,
+        bootstrapCommit.meta.cookieJSON,
+        bootstrapCommit.valueHash,
+        bootstrapCommit.indexes,
+      );
```

Comment:
Maybe add a `setLastMutationID` method instead?

Reply:
Not sure how to do this.  Commit and CommitData are both immutable.